components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1074,7 +1070,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const user = await this.checkAndBlockUser("updateWorkspaceUserPin");
 
         await this.workspaceDb.trace(ctx).transaction(async (db) => {
-            const ws = await this.internalGetWorkspace(user, workspaceId, db);
+            const ws = await this.workspaceService.getWorkspace(user.id, workspaceId);
```

Comment:
just noticed the blunder here, fixed in a follow-up pr: https://github.com/gitpod-io/gitpod/pull/18412/files#diff-f1ba49057ebaf6b6865bb4b7b821d375633ec03f5af573d29d8d70c593e11261R85-R93

Reply:
I don'T even see why this should run in a transaction. There is a single write at the end.

Reply:
This is because we don't want to overwrite any concurrent writes, and because we don't have any workspaceUpdatePartial.

Reply:
It is just one write operation, so it doesn't help with concurrent writes. It is already atomic.